### PR TITLE
Broke out angular-cli docs and added required section about dependencies

### PR DESCRIPTION
**Pull request by pontusr**
It's looking like angular-cli and the ngrx router will be a common combination. I broke out the angular-cli section from the SystemJS file and also added what I believe is a neccessary step (for now) to `npm install´ the dependencies.

### CODE REVIEW

**Comment by brandonroberts** [comment]
Thanks for the PR! There isn't a requirement to `npm install` the dependencies separately as `path-to-regexp` is a dependency of `ngrx/router` and it brings the other dependencies with it.

**Comment by MikeRyan52** [comment]
@brandonroberts I've been given to understand that there is a path-to-regexp conflict between ngrx/router and the CLI, requiring developers to explicitly install that dependency.

**Comment by brandonroberts** [comment]
Yes, but `path-to-regexp` and `query-string` are the only explicit dependencies. We shouldn't need to add every sub-dependency also.

**Comment by pontusr** [comment]
I also feel that installing the explicitly installing the dependencies is not the right way to go. However, I'm not well-versed with SystemJS/Broccoli/Angular CLI to pinpoint the root cause.

The documented step in the PR to install the dependencies using NPM solves the issue however.

Steps to repro the issue with combining angular-cli and ngrx/router according to current docs:

1. Install angular-cli according to https://github.com/angular/angular-cli/ :
```Bash
npm install -g angular-cli
```

2. Generate, build for dev and serve:
```Bash
ng new ngrxtest
cd ngrxtest
ng serve
```
Confirm app us building and working without errors.

3. Build for production and serve:
```Bash
ng serve -prod
```
Confirm app is building and working without errors

4. Install ngrx/router according to https://github.com/nxrx/router/ :
```Bash
npm install @ngrx/router @ngrx/core --save
```

5. Configure CLI project according to https://github.com/ngrx/router/blob/master/docs/overview/systemjs.md#angular-cli-configuration

6. Setup the routing according to the example on https://github.com/nxrx/router/

7. Build and serve dev:
```Bash
ng serve
```
Build works fine but browser console gives:
`zone.js:101 GET http://localhost:4200/vendor/query-string/index.js 404 (Not Found)`

8. Build for production and serve:
```Bash
ng serve -prod
```
Build error: `Error on fetch for vendor/query-string/index.js at file:///E:/Source/ngrxtest/tmp/bundle_plugin-input_base_path-C198oAaa.tmp/0/vendor/query-string/index.js`

Installing query-string using NPM will then render the same error for the next dependency until you have installed all dependencies using NPM - then it works.

**Comment by brandonroberts** [comment]
@pontusr can you share a little bit about your environment? I'm not seeing the same issue on OSX 10.10 with node 5.5.0 and npm 3.5.3

File: docs/overview/angular-cli.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Angular CLI Configuration
+`Angular-CLI` is a cli for making Angular 2 applications. In order to integrate `ngrx/router` into a CLI project, we must install dependencies using NPM and edit the CLI project configuration.
+
+## Dependency installation
+Install the dependencies into your CLI project via npm:
+
+```
+npm install path-to-regexp isarray query-string strict-uri-encode object-assign
```
Comment (brandonroberts) [inline docs/overview/angular-cli.md]: Add `--save` to this line

File: docs/overview/angular-cli.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+The `system-config.ts` and `angular-cli-build.js` files must be updated for `ngrx/router`.
+
+### system-config.ts
+
+The `map` and `packages` sections must be updated in the `system-config.ts` file.
+
+```js
+/** Map relative paths to URLs. */
+const map: any = {
```
Comment (brandonroberts) [inline docs/overview/angular-cli.md]: Go ahead and clean up the quotes to make them all single quotes.

**Comment by pontusr** [comment]
I had this issue on Windows 10, node 4.4.5, npm 2.15.5. After upgrading to node 6.2.1 with npm 3.9.3 I do not see the issue any more.

**Comment by pontusr** [comment]
I fixed the quotes, removed the section about the installation of the npm dependencies as it doesn't seem to be needed and made the instructions more logical and complete (IMO).

**Comment by pontusr** [comment]
Some further testing on another project makes me believe that there is something with regards to the dependencies that ngrx/router needs that require node >4.4.5 and/or npm >2.15.5 - at least on Windows 10. Further testing would be needed unless someon knows which explicit feature/issue in node/npm is required. After pinpointing the requirement this should probably go into readme.md.

**Comment by brandonroberts** [comment]
Thanks for providing that information. It makes more sense why you would need to explicitly install those dependencies also if you're still on npm < 3.x. We'll look at adding a section to the readme for different npm versions.

**Pull request merged**